lib/kwtool_web/controllers/keyword_controller.ex

```diff
@@ -3,15 +3,21 @@ defmodule KwtoolWeb.KeywordController do
 
   alias Kwtool.Crawlers
 
-  def index(conn, _params) do
-    keywords = Crawlers.list_keywords()
+  def index(conn, params) do
+    {keywords, pagination} = Crawlers.get_all_user_keywords(conn.assigns.current_user, params)
```

Comment:
I think we could have something like `list_user_keywords` or `paginate_user_keywords`, what do you think?

Reply:
I also considered about this, however there are 2 methods:
- Query a keyword list of a user (`get_all_user_keywords(%User{} = user, params \\ %{})`)
- Query a particular keyword of a user  (`get_user_keyword(%User{} = user, keyword_id)`)
https://github.com/byhbt/kwtool/blob/617a9e1219314135829be81639abd81c3af7ccfd/lib/kwtool/crawlers.ex#L29-L40

So for example, if we use `list_user_keywords`, then the query single keyword could be `list_user_keyword`, to maintain the consistency, however, I feel it a bit confuse and easy to make mistake for the _single_ vs _plural_ word. (`list_user_keywords` vs `list_user_keyword`)

How about your idea for this case? 🤔

Reply:
I feel like we could use `list_user_keywords` (plural) to retrieve multiple records and `get_user_keyword` (singular) to retrieve a single record. But it's just my preference. I think your current one is also look good to me.

For paginated, I'm thinking of using that name to replace the listing function. I think it would be clear when we use that it is going to return a paginated list and we should get the result of `{keywords, pagination}` instead of just a list of keywords.

But if all of your lists are going to be paginated, it's also ok to go with `list_user_keywords` or `get_user_keywords` or `get_keywords_by_user` 😆 👍

Reply:
Temporarily, I removed the `get_all_user_keywords`, and use the `paginate_user_keywords` instead. 
- `paginate_user_keywords` seems more appropriate in this case.

For querying a single keyword, I will keep `get_keyword_by_user`.

Thanks for the very cool feedback 😄  👍